Docstrings improved in calculus/singularities.py

AnimeshSinha1309 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
Partially improves #15370 - Only for the file calculus/singularities.py

#### Brief description of what is fixed or changed
* Changed all the doctrings in the module calculus.singularities to the numpydoc format.

#### Release Notes
<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

AnimeshSinha1309 committed:
Improved Docstrings of calculus/singularities.py

Got all docstrings to numpydoc format and changed two variable name to get a perfect pylint score.

AnimeshSinha1309 committed:
Fixed the errors in monotonicity_helper docstrings

Did not change the function name to monotonicity_helper in the previous commit, did it this time.

AnimeshSinha1309 committed:
Removed monotonicity_helper examples

monotonicity_helper was a private method that cannot be accessed, so removing the docstring examples I added.

moorepants commented:
This looks great. +1 from me if the tests pass.

moorepants commented on doc/README.rst:
```diff
@@ -16,6 +16,10 @@ and to view it, do::
+If you get **mpmath** error, install python3-mpmath package::
+
+    apt-get install python3-mpmath
+
```
Please move these to a different PR.

AnimeshSinha1309 replied:
Sure will do, don't know how to move them out but I guess I can make another commit undoing these changes, and then redo them from another branch.

moorepants commented on README.rst:
```diff
@@ -63,6 +63,9 @@ You can generate everything at the above site in your local copy of SymPy by::
 
+To fulfill the package requirement, ensure that all the dependencies for building
+documentation enlisted in `doc/README.rst` are satisfied.
+
```
move to different PR

AnimeshSinha1309 replied:
Okay, did move both these files to #15626, Deleted those commits here. Sorry for the force push, but knew no other way.

moorepants commented on sympy/calculus/singularities.py:
```diff
@@ -44,6 +53,21 @@ def singularities(expression, symbol):
+    Set
+        A set of values for `symbol` for which `expression` has a
+        singularity. An `EmptySet` is returned if `expression` has no
+        singularities for any given value of `Symbol`.
```
Note that in RestructuredText if you want monospaced font you need to do: ` ``my text`` `

AnimeshSinha1309 committed:
Double ticks in RST Monospaced Code

Had single ticks last time, as in Markdown. Fixed that here.

AnimeshSinha1309 commented:
@moorepants can you please check and tell me if I am done with everything. It this ready to be merged?

moorepants commented on sympy/calculus/singularities.py:
```diff
@@ -28,6 +28,15 @@ def singularities(expression, symbol):
     Currently supported functions are:
     - univariate rational (real or complex) functions
```
Things like this can be in the "Notes" section below.

moorepants commented on sympy/calculus/singularities.py:
```diff
@@ -189,6 +297,13 @@ def is_strictly_decreasing(expression, interval=S.Reals, symbol=None):
+
+    Boolean
+        True if ``expression`` is strictly decreasing in the given ``interval``,
+        False otherwise.
```
Examples should come after "Returns". The first two sections should be "Parameters" and "Returns". I think the numpydoc spec gives the best ordering of sections.

moorepants commented on sympy/calculus/singularities.py:
```diff
@@ -197,6 +312,13 @@ def is_monotonic(expression, interval=S.Reals, symbol=None):
+        The target function which is being checked.
+    interval: Symbol
+        The range of values in which we are testing, defaults to all reals.
+    symbol: Symbol
```
One thing to check is whether this strictly has to be a Symbol. Often a Function might work or even sometimes an expression. I don't know how restrictive these functions are but many sympy functions aren't as restrictive. So try to check the restrictions of input type and then you can do "Symbol or Function" or "sympyfiable" if sympify is called internal to the function.

AnimeshSinha1309 committed:
Order of Subsections and Return Types

Changed the order of subsections, spacing, and added optional parameters in docstrings to match numpydoc format.

AnimeshSinha1309 commented:
I have fixed the order of items, and moved the thing to Notes. I think I got the return types right.
All expressions we are checking have to be `Expr`, does not make sense otherwise. And then interval has to be a `Set`, that is the most flexible abstract class *(changed this one)*. And finally symbols, I don't see how they can be anything but `Symbol`, @moorepants please tell me if I am wrong on this.

moorepants commented:
Looks great. Thanks for the fix!
